src/braket/experimental/autoqasm/api/api.py

```diff
@@ -50,9 +50,6 @@ def function(f: Callable) -> Callable[[Any], aq_program.Program]:
         Callable[[Any], Program]: A callable which returns the converted
         quantum program when called.
     """
-    if is_autograph_artifact(f):
-        return f
-
```

Comment:
I think we could hit this code by calling a function which is double-decorated with `@aq.function`. For example, something like:
```
@aq.function
def main():
    pass
program = aq.function(main)()
```
That's contrived obviously, but in general it might be nice to keep this behavior by adding a test to cover that case.

Reply:
In this case (double decorated), we still want to transform the function but not return it directly?